### DISCUSSION TITLE
EXITING because of FATAL ERROR in reads input: short read sequence line: 0

### DISCUSSION THREAD

**Discussion by pd1305** [alexdobin/STAR#1225] [Q&A]
Hello!

I have been trying to run star on mouse RNAseq paired end samples. The first time I ran the following command and I got an error.
Command
STAR --runThreadN 16 --genomeDir /repositories/mm10 --readFilesIn /Data/File_1.fq /Data/File2_2.fq --outFilterType BySJout --outFilterMultimapNmax 20 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.6 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --outSAMattributes NH HI NM MD --outSAMtype BAM SortedByCoordinate --outFileNamePrefix Test_1

Error: "EXITING because of FATAL ERROR in reads input: short read sequence line: 0"
Read Name=@A00685:109:HVW3JDMXX:1:1101:14226:1329
Read Sequence====
DEF_readNameLengthMax=50000
DEF_readSeqLengthMax=650

Final output:
             Started job on |	Apr 30 10:33:06
                             Started mapping on |	Apr 30 10:33:23
                                    Finished on |	Apr 30 10:36:44
       Mapping speed, Million of reads per hour |	847.75

                          Number of input reads |	47332564
                      Average input read length |	202
                                    UNIQUE READS:
                   Uniquely mapped reads number |	0
                        Uniquely mapped reads % |	0.00%
                          Average mapped length |	0.00
                       Number of splices: Total |	0
            Number of splices: Annotated (sjdb) |	0
                       Number of splices: GT/AG |	0
                       Number of splices: GC/AG |	0
                       Number of splices: AT/AC |	0
               Number of splices: Non-canonical |	0
                      Mismatch rate per base, % |	-nan%
                         Deletion rate per base |	0.00%
                        Deletion average length |	0.00
                        Insertion rate per base |	0.00%
                       Insertion average length |	0.00
                             MULTI-MAPPING READS:
        Number of reads mapped to multiple loci |	438764
             % of reads mapped to multiple loci |	0.93%
        Number of reads mapped to too many loci |	6834
             % of reads mapped to too many loci |	0.01%
                                  UNMAPPED READS:
       % of reads unmapped: too many mismatches |	0.00%
                 % of reads unmapped: too short |	98.99%
                     % of reads unmapped: other |	0.07%
                                  CHIMERIC READS:
                       Number of chimeric reads |	0
                            % of chimeric reads |	0.00%

After reading old posts, I tried to play with some filters but the uniquely mapped reads stayed 0%, whereas multiple mapped reads became 99%

--outFilterMismatchNoverLmax 0.3 --outFilterScoreMinOverLread 0 --outFilterMatchNminOverLread 0 

Started job on |	Apr 30 11:31:00
                             Started mapping on |	Apr 30 11:31:31
                                    Finished on |	Apr 30 11:40:29
       Mapping speed, Million of reads per hour |	316.72

                          Number of input reads |	47332564
                      Average input read length |	202
                                    UNIQUE READS:
                   Uniquely mapped reads number |	52
                        Uniquely mapped reads % |	0.00%
                          Average mapped length |	98.71
                       Number of splices: Total |	81
            Number of splices: Annotated (sjdb) |	80
                       Number of splices: GT/AG |	80
                       Number of splices: GC/AG |	1
                       Number of splices: AT/AC |	0
               Number of splices: Non-canonical |	0
                      Mismatch rate per base, % |	1.38%
                         Deletion rate per base |	0.04%
                        Deletion average length |	2.00
                        Insertion rate per base |	2.38%
                       Insertion average length |	4.21
                             MULTI-MAPPING READS:
        Number of reads mapped to multiple loci |	47200528
             % of reads mapped to multiple loci |	99.72%
        Number of reads mapped to too many loci |	99953
             % of reads mapped to too many loci |	0.21%
                                  UNMAPPED READS:
       % of reads unmapped: too many mismatches |	0.00%
                 % of reads unmapped: too short |	0.00%
                     % of reads unmapped: other |	0.07%
                                  CHIMERIC READS:
                       Number of chimeric reads |	0
                            % of chimeric reads |	0.00%

I have used original fastq files before adapter trimming and after adapter trimming and still had the same error. I have tried running STAR only in single end mode and it still didn't work.

This is the output when I grep the fastq file:
grep -A10 ^@A00685:109:HVW3JDMXX:1:1101:14226:1329 File.fastq
@A00685:109:HVW3JDMXX:1:1101:14226:1329 1:N:0:CCGCGGTT+CTAGCGCT
GATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGCGGTTATCGCGTATGCCGTCTTCTGCTTGAAGATGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG
+
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFF:,:FF:F:F::FFFF,::,::,::,,,,,::FF:F:FFFFFFFFFFFFFFFFFFFFFF
@A00685:109:HVW3JDMXX:1:1101:16468:1329 1:N:0:CCGCGGTT+CTAGCGCT
GTCCCAGAGACAGGACTCCGGCCATCAGACTTGGAAGCCAGCACCTCTACCAGCTGAATCATCTTGCCTGCCTTAAAGTGAATTTCCTTAAGAGGGAGAAT
+
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FF
@A00685:109:HVW3JDMXX:1:1101:16721:1329 1:N:0:CCGCGGTT+CTAGCGCT
TGGTAAGGCTTTCTGTGAAAAACACAAATTCCGGAACTTCATAGCTCATTAGTGTTGCTTTCTTCTAAATGAACAGCCTTCTGTGAGCCTGGCTTTTCCTC
+

Any pointers in this direction will be very helpful to me. Thank you very much.
Cheers,
Priya

**Comment by alexdobin** [comment]
Hi Priya,

this is more suitable as an Issue than Discussion.
This indicates some issues with the FASTQ files, such as invisible control characters at the end of strings.
What version are you using? Please try the latest one (2.7.8a or 2.7.9a) as they may deal better with these issues.

If this does not help, please grep this read with grep -A3 from both files into R1.fq and R2. fq, and try to map just the resulting files.

Cheers
Alex
